### ISSUE TITLE
Migrate to NextJS?

### ISSUE THREAD

**Issue by Hacksore** [MelkeyOSS/astrostation#265]
Benefits:
- we can have an API

Drawbacks
- Lots of work

**Comment by Melkeydev** [comment]
The drawbacks

**Comment by Melkeydev** [comment]
On a serious note - @royanger and I have been SLOWLY talking about making a BE for this.

**Comment by royanger** [comment]
I don't think we need Next. We aren't going to use page based routing, SSR or SSG, etc, e tc. The only benefit is the API route/serverless, but if we want to build the backup with serverless we can handle that other ways.

**Comment by Hacksore** [comment]
> I don't think we need Next. We aren't going to use page based routing, SSR or SSG, etc, e tc. The only benefit is the API route/serverless, but if we want to build the backup with serverless we can handle that other ways.

🧢.

Your best bet is to use next for the fremium tier.

Trying to roll your own backend is going to have cost and scaling problem.

If you have a compelling reason to build a fully custom solution so be it but personally I find next to be the most low effort/cost solution.

**Comment by royanger** [comment]
There are definitely some free options out there. I believe that Melkey is already paying on Vercel, but that's not really an issue either way. We don't need Next to use serverless on Vercel if we want to use serverless on Vercel. We can add serverless to any framework, and the Vercel CLI can even provide a dev server for a nice DX.

**Comment by Hacksore** [comment]
Closed in favor of https://github.com/MelkeyOSS/astrostation/issues/238, the vercel serverless looks like a good option for API routes.

However, there is still an issue of requiring a database but I'll let the tech leads figure that one out (i just work here).
